### PR TITLE
Add a classification model based on seq2vec

### PR DESCRIPTION
**Pull request by acohan**
- add a dataset reader for generic text classification tasks
- add a classification model that encodes the text using a seq2vec encoder and then uses a FF on top to classify the results
- add sample data from citation classification

### CODE REVIEW

File: allennlp_config/citation_classification.jsonnet [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "random_seed": std.parseInt(std.extVar("SEED")),
+  "pytorch_seed": std.parseInt(std.extVar("PYTORCH_SEED")),
+  "numpy_seed": std.parseInt(std.extVar("NUMPY_SEED")),
+  "dataset_reader": {
+    "type": "classification_dataset_reader",
+  },
+  "train_data_path": std.extVar("CITATION_TRAIN_PATH"),
```
Comment (ibeltagy) [inline allennlp_config/citation_classification.jsonnet]: if you want exp.sh to be usable for all config files, then path here and in other config files will need to change. Maybe this is a different PR

File: allennlp_config/citation_classification_bert.jsonnet [hunk 2]
```diff
@@ -0,0 +1,86 @@
+{
```
Comment (ibeltagy) [inline allennlp_config/citation_classification_bert.jsonnet]: are we aiming to use the same config file for PICO as well ? if so, then maybe rename the file to a more generic name, like `classification_bert.jsonnet`

File: allennlp_config/citation_classification_bert.jsonnet [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          "do_lowercase": std.extVar("is_lowercase"),
+          "use_starting_offsets": true
+      },
+      "token_characters": {
+        "type": "characters",
+        "min_padding_length": 3
+      }
+    }
+  },
+  "train_data_path": std.extVar("NER_TRAIN_DATA_PATH"),
```
Comment (ibeltagy) [inline allennlp_config/citation_classification_bert.jsonnet]: same here.

File: sci_bert/models/text_classifier.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                 classifier_feedforward: FeedForward,
+                 verbose_metrics: False,
+                 initializer: InitializerApplicator = InitializerApplicator(),
+                 regularizer: Optional[RegularizerApplicator] = None,
+                 ) -> None:
+        super(TextClassifier, self).__init__(vocab, regularizer)
+
+        self.text_field_embedder = text_field_embedder
+        self.num_classes = self.vocab.get_vocab_size("labels")
+        print("num_classes: ", self.num_classes)
```
Comment (ibeltagy) [inline sci_bert/models/text_classifier.py]: nit: remove this or use log

File: allennlp_config/citation_classification.jsonnet (hunk 1 above)
Comment (acohan) [reply allennlp_config/citation_classification.jsonnet]: done

File: scripts/exp_citation.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
```
Comment (ibeltagy) [inline scripts/exp_citation.sh]: it will be great if we use exp.sh. Again, another PR with the suggestions mentioned above.

Comment (acohan) [reply scripts/exp_citation.sh]: yes, I'll remove this to keep the codebase cleaner.

**Comment by acohan** [comment]
addressed comments.

**Pull request merged**